src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -55,13 +57,36 @@ static HttpKernel()
                                  This Kernel is able to execute http requests and display the results.
                                  """;
 
-        _client = client ?? new HttpClient();
+        _client = client ?? new HttpClient() { Timeout = Timeout.InfiniteTimeSpan };
         _responseDelayThresholdInMilliseconds = responseDelayThresholdInMilliseconds;
         _contentByteLengthThreshold = contentByteLengthThreshold;
 
         RegisterForDisposal(_client);
     }
 
+    /// <summary>
+    /// Set a timeout for HTTP requests that are issued using this <see cref="HttpKernel"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// If an <see cref="HttpClient"/> is not passed in the constructor, the specified
+    /// <paramref name="requestTimeoutInMilliseconds"/> will be the only timeout that applies.
+    /// </para>
+    /// <para>
+    /// If an <see cref="HttpClient"/> is passed in the constructor, the timeout specified in
+    /// <see cref="HttpClient.Timeout"/> will not be overridden by the specified
+    /// <paramref name="requestTimeoutInMilliseconds"/>. The shorter of the two timeouts will apply in such cases.
+    /// </para>
+    /// <para>
+    /// Note that <see cref="HttpClient.Timeout"/> has a default value of 100 seconds. If you wish to control the
+    /// timeout exclusively via <paramref name="requestTimeoutInMilliseconds"/>, either omit the 
+    /// <see cref="HttpClient"/> parameter in the constructor, or pass in an <see cref="HttpClient"/> with
+    /// <see cref="HttpClient.Timeout"/> set to <see cref="Timeout.InfiniteTimeSpan"/>.
+    /// </para>
+    /// </remarks>
+    public void SetRequestTimeout(int requestTimeoutInMilliseconds)
```

Comment:
It seems like this should be a property with both a getter and a setter and, given the usage below, changed to a `TimeSpan`.